File: Mail/DataToKafka/src/WorkTask.java
/**
* Copyright @ 2015 ShanghaiKunyan. All rights reserved
* @author     : Sunsolo
* Email       : wukun@kunyan-inc.com
* Date        : 2016-04-08 17:09
* Description : 
*/

package com.kunyan.wokongsvc.mail;

import com.kunyan.wokongsvc.mail.DateHandle;
import com.kunyan.wokongsvc.mail.WorkSource;

import java.util.TimerTask;              
import java.util.Calendar;
import java.text.SimpleDateFormat;

/**
 * 定时邮件数据的发送(发到kafka)都在这
 **/
class WorkTask extends TimerTask {
  private DateHandle dateHandle = new DateHandle(new SimpleDateFormat("yyyyMMdd"));

  WorkTask() {
  }

  static {
  }

  @Override
  public void run() {
    Calendar cal = Calendar.getInstance();
    WorkSource workSource = new WorkSource(dateHandle.nowMinute(cal), dateHandle.nowTime(cal));
    workSource.init();
    workSource.sendStockEmailJson();
    workSource.sendIndustryEmailJson();
    workSource.sendSectionEmailJson();
  }
}

File: SHRealData/src/WorkTask.scala
/*=============================================================================
#    Copyright (c) 2015
#    ShanghaiKunyan.  All rights reserved
#
#    Filename     : /home/wukun/SparkKafka/src/main/scala/WorkTask.scala
#    Author       : Sunsolo
#    Email        : wukun@kunyan-inc.com
#    Date         : 2016-04-14 10:54
#    Description  : 
=============================================================================*/

package com.kunyan.wokongsvc.realtimedata

import com.kunyan.wokongsvc.realtimedata.WorkHandle

import org.apache.log4j.Logger

import java.lang.ClassNotFoundException
import java.sql.SQLException

class WorkTask(xmlHandle:XmlHandle) {
  val work:WorkHandle = WorkHandle(xmlHandle)
  val infoLogger:Logger = Logger.getLogger(classOf[InfoLogger])
  val warnLogger:Logger = Logger.getLogger(classOf[WarnLogger])

  def run() {
    try {
      work.initData
      infoLogger.info("StockCode initial success")
    } catch {
      case e:SQLException => {
        warnLogger.warn("SQLException:mysql exception")
      }
      case e:ClassNotFoundException => {
        warnLogger.warn("com.mysql.jdbc.Driver is not found")
      }
    }
  }
}

object WorkTask {

  def apply(xmlHandle:XmlHandle):WorkTask = {
    new WorkTask(xmlHandle)
  }
}

